apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,56 @@
     this.logger.debug('Clock updated', now);
   };
 
+  LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+
+      // if there is no alarm
+      if(!alarmData.hour && !alarmData.minute) {
+        this.resources.elements.alarm.classList.add('no-alarms');
+        return;
+      }
+
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.resources.elements.alarm.classList.add('has-alarm');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    }).catch(() => {
+      this.resources.elements.alarm.classList.remove('has-alarm');
+      this.resources.elements.alarm.classList.add('no-alarms');
+    });
+	};
+
+  /*
+   * Gets the Alarm Data from the DataStore.
+   */
+  LockScreenClockWidget.prototype.fetchAlarmData = function() {
+    return new Promise((resolve, reject) => {
+      navigator.getDataStores('alarms')
+      .then((stores) => {
+        return [stores[0].getLength(), stores[0]];
+      }).then(([len, store]) => {
+        if(0 === len) {
+          return;
+        }
+        return store.get(1);
+      }).then((fetchedData) => {
```

Comment:
Here, if you fetch no data, you can design this `then` callback as:

    .then((fetchedData = null) => {..... resolve(fetchedData)})

So you can check it in line 80, rather than to throw exception and catch it.